be/src/exprs/vectorized/time_functions.cpp

```diff
@@ -488,6 +488,113 @@ DEFINE_UNARY_FN_WITH_IMPL(week_of_yearImpl, v) {
 }
 DEFINE_TIME_UNARY_FN(week_of_year, TYPE_DATETIME, TYPE_INT);
 
+uint TimeFunctions::week_mode(uint mode) {
+    uint week_format = (mode & 7);
+    if (!(week_format & WEEK_MONDAY_FIRST)) week_format ^= WEEK_FIRST_WEEKDAY;
+    return week_format;
+}
+
+/*
+   Calc days in one year.
+   @note Works with both two and four digit years.
+   @return number of days in that year
+*/
+uint TimeFunctions::compute_days_in_year(uint year) {
+    return ((year & 3) == 0 && (year % 100 || (year % 400 == 0 && year)) ? 366 : 365);
+}
+
+/*
+   Calc weekday from daynr.
+   @retval 0 for Monday
+   @retval 6 for Sunday
+*/
+int TimeFunctions::compute_weekday(long daynr, bool sunday_first_day_of_week) {
+    return (static_cast<int>((daynr + 5L + (sunday_first_day_of_week ? 1L : 0L)) % 7));
+}
+
+/*
+  Calculate nr of day since year 0 in new date-system (from 1615).
+  @param year	  Year (exact 4 digit year, no year conversions)
+  @param month  Month
+  @param day	  Day
+  @note 0000-00-00 is a valid date, and will return 0
+  @return Days since 0000-00-00
+*/
+long TimeFunctions::compute_daynr(uint year, uint month, uint day) {
+    long delsum;
+    int temp;
+    int y = year;
+
+    if (y == 0 && month == 0) {
+        return 0;
+    }
+
+    delsum = static_cast<long>(365 * y + 31 * (static_cast<int>(month) - 1) + static_cast<int>(day));
```

Comment:
is it possiable y != 0 and month = 0

Reply:
month = 0 means this date value is illegal, so input vlaue is null, this logic willn't be executed.

Reply:
now week_of_year_with_default_modeImpl was define as VectorizedStrictBinaryFunction
which means the input for invalid date is also possiable

Reply:
> now week_of_year_with_default_modeImpl was define as VectorizedStrictBinaryFunction which means the input for invalid date is also possiable

The input date must be valid here, because any illegal input can't obtain date type here(Actually, That's null), and that follow another logic. And legal date willn't produce illegal output, So we needn't add check here.